ui/app/components/auth/form-template.ts

```diff
@@ -77,8 +67,33 @@ export default class AuthFormTemplate extends Component<Args> {
 
   constructor(owner: unknown, args: Args) {
     super(owner, args);
-    this.selectedAuthMethod = this.args.preselectedAuthType || 'token';
-    this.fetchMounts.perform();
+    // ensures args have settled before setting form state
+    setTimeout(() => this.initializeState(), 0);
```

Comment:
Another way to approach this without using setTimeout is @tracked properties. 
```suggestion
@tracked @stateInitialized = false;

get isReadyToInitialize() {
 return this.args.anArg !== undefined;
}

constructor(owner, args) {
    super(owner, args);
    if (this.isReadyToInitialize) {
      this.initializeState();
    }
  }
  
  initializeState() {
    this.stateInitialized = true;
    ... all the other code
  }
```
`setTimeout(...,0)` as I understand it, says to the event loop wait until the next tick. In this case, it means that the args passed in must be ready by the next tick, which may not be true. It also doesn't respond to changes in the args, so if the data were to change the state doesn't get re-initialized. And lastly (this is why setTimeout is on my mind) it makes testing a little harder. Not super hard, but it may require us to throw in an `await settled()`.

Reply:
One last thought, if the args coming in are async you can use [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame), which tells the browser to run whatever you pass into it right before the next repaint of the screen. In practice, this means `setTimeout` runs as soon as the stack is cleared, and `requestAnimationFrame` waits until the next frame is drawn.

Reply:
For posterity, refactored to use `did-insert`